tests/test_core.py

```diff
@@ -17,6 +17,7 @@
 import scipy.io
 import six
 import pytest
+import unittest.mock
```

Comment:
This import fails on python 2.  Please use the `six.moves` alias for this until we can drop py2.7 support in the 0.8 series.

Reply:
I didn't see a `six.moves` alias for `mock`. (I looked [here](https://six.readthedocs.io/#module-six.moves) in case I missed it.) `mock` was a [separate package](https://pypi.org/project/mock/) in Python 2 that was added to the standard library in Python 3. I could add `mock` to the tests requirements and use `import mock` instead?

Reply:
That's strange -- I would expect six to have an alias in this case.

As an alternative, you could do the following:

```python
try:
    import mock
except ImportError
    from unittests import mock
```
which would work for our purposes here.  I'm not used to using mock for this kind of test though -- is it necessary in your opinion?  Or could there be a more direct way to test this?

Reply:
As is, `reassigned_spectrogram` only does clean-up on the results of the two support functions. My preference is to mock those two functions so that the tests for `reassigned_spectrogram` can just test its functionality separately, without depending on the correctness of the support functions.

It could be tested using data that is known to violate the reassignment assumptions, where some of the numerical estimates from `reassign_*` are expected to be NaN or outside the spectrogram limits. But those estimates are the most likely to be unstable if the support functions change in the future.

I ran into this when I compared this implementation to the [tfr](https://github.com/bzamecnik/tfr) package, which uses a different reassignment algorithm. Because of that, I prefer mocking the support functions to have fixed return values, and only testing that `reassigned_spectrogram` does the right thing with those values. What do you think?

Reply:
Ok, that's reasonable, thanks for the explanation.  I think we should just go with the try..except for py2/3 support, and that will make it easy to transition to 3-only in the 0.8 series.

Reply:
Done in https://github.com/librosa/librosa/pull/926/commits/a444584bf0483490aaa900b94d35f2ec51a137b3